app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -90,8 +103,78 @@ internal fun MessageBody(
         StrikethroughExtension.builder().requireTwoTildes(true).build(),
         TablesExtension.create()
     )
+    text?.also {
+        MarkdownDocument(Parser.builder().extensions(extensions).build().parse(it) as Document, nodeData)
+    }
+    buttonList?.also {
+        MessageButtonsContent(
+            messageId = messageId,
+            buttonList = it,
+            onClick = onButtonClick,
+            pendingButton = pendingButton
+        )
+    }
+}
+
+@Composable
+fun MessageButtonsContent(
+    messageId: String,
+    pendingButton: String?,
+    buttonList: List<MessageButton>,
+    onClick: ((messageId: String, buttonId: String) -> Unit)?
+) {
+    Column(
+        modifier = Modifier
+            .wrapContentSize()
+    ) {
+
+        for (index in buttonList.indices) {
+            val button = buttonList[index]
+            MessageButtonItem(
+                messageId = messageId,
+                pendingButtonId = pendingButton,
+                button = button,
+                onClick = onClick
+            )
+            if (index != buttonList.lastIndex) {
+                Spacer(modifier = Modifier.padding(top = dimensions().spacing8x))
+            }
+        }
+    }
+}
+
+@SuppressLint("RememberReturnType")
+@Composable
+fun MessageButtonItem(
+    messageId: String,
+    button: MessageButton,
+    pendingButtonId: MessageButtonId?,
+    onClick: ((messageId: String, buttonId: String) -> Unit)?
+) {
+    val onCLick = remember(button.isSelected) {
```

Comment:
here would be a good place for that single ViewModel I think, each MessageButtonItem, will just execute the use case and hold the pending state